capytaine/bodies/bodies.py

```diff
@@ -64,6 +66,41 @@ def __init__(self, mesh=None, dofs=None, name=None):
 
         LOG.info(f"New floating body: {self.name}.")
 
+    @staticmethod
+    def from_meshio(mesh, name=None, keep_immersed_part=True, remove_degenerated_faces=True) -> 'FloatingBody':
+        """Create a FloatingBody from a meshio mesh object."""
+
+        import meshio
+        if not isinstance(mesh, meshio._mesh.Mesh):
+            raise TypeError('mesh must be of type meshio._mesh.Mesh, recevied {:}'.format(type(mesh)))
+
+        if name is None:
+            date_str = datetime.datetime.now().strftime('%Y%m%d%H%M%S%f')
+            name = 'fb_{:}'.format(date_str)
+
+        def all_faces_as_tetra(cells):
+            all_faces = []
+            if 'tetra' in cells:
+                all_faces.append(cells['tetra'])
+            if 'triangle' in cells:
+                triangles_as_tetra = np.empty((cells['triangle'].shape[0], 4), dtype=int)
+                triangles_as_tetra[:, :3] = cells['triangle'][:, :]
+                triangles_as_tetra[:, 3] = cells['triangle'][:, 2]  # Repeat one node to make a tetra
+                all_faces.append(triangles_as_tetra)
+            return np.concatenate(all_faces)
+
+        cpt_mesh = Mesh(vertices=mesh.points, 
+                        faces=all_faces_as_tetra(mesh.cells_dict))
+
+        fb = FloatingBody(mesh=cpt_mesh, name=name)
+        if keep_immersed_part:
+            fb.keep_immersed_part()
+        if remove_degenerated_faces:
+            fb.mesh.remove_degenerated_faces()
```

Comment:
yes, I think that might be good. When I was doing this, I was considering having a warning message along the lines of

```python
warn('removed {:} bad cells'.format(num_bad_cells))
```

just so that this doesn't go unnoticed.

Want me to take a crack at address this specifically in a separate PR addressing #46?

Reply:
I had drafted something there: 38e2119, but you're right, it is missing some feedback when the mesh is modified.

Reply:
I was thinking about this some more... I would propose a separate PR (that will be completed before this one) that updates the instantiation of a `FloatingBody` to

1. By default, execute `keep_immersed`
2. By default, execute `heal_mesh`

Additionally, I think we should considering updating the functions called by `heal_mesh` so that they report the changes they make (e.g., number of faces removed) as `LOG.warning` or `LOG.info` (currently done with `LOG.debug`).

Reply:
> By default, execute keep_immersed

I don't know about that.
I kept it separate because I assumed people might want to preprocess the mesh first (e.g. translate it vertically) and clipping should always be the last thing. Changing the default behavior now would break some code and I'm not sure it's worth it.  Maybe automatically clipping the mesh when initializing a new RadiationProblem/DiffractionProblem instead?

> By default, execute heal_mesh

I working on that on the `cylinder` branch (maybe I should open a PR to make it more visible). `heal_mesh` was not defined for some symmetric meshes so it'll still need some testing before merging.

> Additionally, I think we should considering updating the functions called by heal_mesh so that they report the changes they make

Sure, I'm open to any proposal to improve these functions. They were copied almost exactly from meshmagick, so you can also submit the patches there.

Reply:
This all sounds good. It sounds like we should probably wait on this PR until you merge in your work on the `cylinder` branch.

Reply:
Automatic mesh healing has been merged with #83. I'm removing it from this PR.

I'll also remove the automatic mesh clipping: mostly because it is confusing if `from_meshio` is the only `FloatingBody` constructor that does it without notice. But I'm not against adding an automatic clipping somewhere in another PR someday.